ethers-solc/src/artifacts.rs

```diff
@@ -379,28 +382,38 @@ impl CompilerOutput {
         self.errors.iter().any(|err| err.severity.is_error())
     }
 
-    pub fn diagnostics(&self) -> OutputDiagnostics {
-        OutputDiagnostics(&self.errors)
+    pub fn diagnostics<'a>(&'a self, ignored_error_codes: &'a [u64]) -> OutputDiagnostics {
+        OutputDiagnostics { errors: &self.errors, ignored_error_codes }
     }
 }
 
 /// Helper type to implement display for solc errors
 #[derive(Clone, Debug)]
-pub struct OutputDiagnostics<'a>(&'a [Error]);
+pub struct OutputDiagnostics<'a> {
+    errors: &'a [Error],
+    ignored_error_codes: &'a [u64],
+}
 
 impl<'a> OutputDiagnostics<'a> {
     pub fn has_error(&self) -> bool {
-        self.0.iter().any(|err| err.severity.is_error())
+        self.errors.iter().any(|err| err.severity.is_error())
     }
 }
 
 impl<'a> fmt::Display for OutputDiagnostics<'a> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         if !self.has_error() {
-            f.write_str("Compiler run successful\n")?;
+            f.write_str("Compiler run successful")?;
         }
-        for err in self.0 {
-            writeln!(f, "{}", err)?;
+        for err in self.errors {
+            // Do not log any ignored error codes
+            if let Some(error_code) = err.error_code {
+                if !self.ignored_error_codes.contains(&error_code) {
+                    writeln!(f, "\n{}", err)?;
```

Comment:
I prefer the newline here so that if there's no errors there's no extra newline in the Compiler run successful log, otherwise it adds an empty line which is redundant